source/api/commands/invoke.md

```diff
@@ -132,6 +132,70 @@ const double = (n) => n * n
 cy.wrap([reverse, double]).invoke(1, 4).should('eq', 16)
 ```
 
+## Invoking an async function
+
+In this example we invoke a Vuex action that returns a Promise. `cy.invoke` will wait until the Promise resolves after the specified delay which we pass as an argument and only then will continue executing. 
+
+A minimal Vuex store:
+```javascript
+function randomId () {
+  return Math.random().toString().substr(2, 10)
+}
+
+const store = new Vuex.Store({
+  state: {
+    todos: []
+  },
+  mutations: {
+    ADD_TODO (state, todoObject) {
+      state.todos.push(todoObject)
+    }
+  },
+  actions: {
+    // example promise-returning action
+    addTodoAfterDelay ({ commit }, { milliseconds, title }) {
+      return new Promise((resolve) => {
+        setTimeout(() => {
+          const todo = {
+            title,
+            completed: false,
+            id: randomId(),
+          }
+
+          commit('ADD_TODO', todo)
+          resolve()
+        }, milliseconds)
+      })
+    },
+  },
+})
+```
+
+The Cypress Test with `cy.inkoke()` awaiting the promise:
+```javascript
+// For this command to work you have to expose your vuex store under window.store for cypress to be able to access it
+// e.g. like this
+// if (isCypress) {
+//    window.store = store
+// }
+const getVuex = () => cy.window({ log: false }).its('store')
+
+getVuex().invoke('dispatch', 'addTodoAfterDelay', {
+  milliseconds: 2000,
+  title: 'async task'
+})
+
+// log message appears after 2 seconds
+cy.log('after invoke')
+
+// assert UI
+getTodoItems().should('have.length', 1).first().contains('async task')
```

Comment:
getTodoItems() is not defined in this example. Can we have an example that does not rely on Vue? Because this is still not copy/pastable. You'd have to require Vuex here. Just a function that resolves a Promise?

If not, I would just link to the example and remove the example inline.

Reply:
You are right.
I found it quite hard but I finally came up with a good example that doesn't contain an empty Promise that just resolves after a timeout.